### ISSUE TITLE
Podium does not inherited email address

### ISSUE THREAD

**Issue by davidnewcomb** [bizley/yii2-podium#107]
I was wondering if the email address would need to come across too and I think it does.
While viewing the forums, the following message keeps (erroneously) appearing.

    No e-mail address has been set for your account! Go to Profile > Account Details to add one

I think that this is a symptom of a bigger problem which is that Podium will fail to email notifications because it doesn't have an email address for an inherited user.

So I think we need another one of those new parameter thingys to cover email address as well.

**Comment by bizley** [comment]
Well, yes... Now that you are speaking about this... Maybe better idea is to replace all the user single fields with one array.

**Comment by davidnewcomb** [comment]
First there was password and user name, now there is email address. There may well be others and so having a container for all of the user inherited properties would simplify the surrounding code.

**Comment by elkrari** [comment]
I had the same problem/question...
You can (if you didn't yet solve it) add `$newUser->email = Yii::$app->user->identity->email;` to `createInheritedAccount()` in the User model. It will copy the email from the main App user session.

**Comment by bizley** [comment]
It will be a part of user fields' array. Your solution is not good since email address field can be named differently.

**Comment by elkrari** [comment]
Yes of course. I'm not giving enhancement but just a "solution" that each one can add himself to the code (and customize) to solve the problem if it's urgently needed.

**Comment by davidnewcomb** [comment]
This issue was the only thing that worried be about releasing. I've added the fix to my branch along with the slug stuff, mini-forum events and now this user hack. It was enough to let me release today.
I'll have to tidy my code when version 0.8 is released but for the moment I now have everything I need.
Thanks @elkrari (and @bizley of course)

**Comment by bizley** [comment]
Nice! Thanks for being patient with my lack of releases lately. Lots of other work, no time for Podium...

But API is coming up and will be sweet 💥

**Comment by davidnewcomb** [comment]
I'm looking forward to the API. There is loads I wanted to do but I have to keep remembering to do my own work ;)

**Comment by elkrari** [comment]
@davidnewcomb Thrilled to help you :)
The API will be a good enhancement to Podium!
I've tried many forum modules for Yii2, and Podium (for me) is best and easiest one to install and configure. Thanks @bizley for the job!

**Comment by davidnewcomb** [comment]
@bizley Is there any news on the "sweet API"? Can you checkin what you have, maybe some of us can help you complete it.

**Comment by bizley** [comment]
I'm afraid there is not much now, my current job is pretty overwhelming :( But please bear with me, this will happen sooner than later.
